bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/GeneratedScriptPreviewPage.java

```diff
@@ -0,0 +1,194 @@
+import org.eclipse.core.databinding.observable.value.IValueChangeListener;
+import org.eclipse.core.databinding.observable.value.ValueChangeEvent;
+import org.eclipse.core.databinding.observable.value.WritableValue;
+import org.eclipse.emf.databinding.EMFObservables;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.source.SourceViewer;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * @author aurelie
+ */
+public class GeneratedScriptPreviewPage extends WizardPage {
+
+    private final WritableValue selectedDataObservable;
+    private final WritableValue rootNameObservable;
+    private final WritableList fieldToContractInputMappingsObservable;
+    private final FieldToContractInputMappingOperationBuilder operationBuilder;
+    private final RepositoryAccessor repositoryAccessor;
+    private final FieldToContractInputMappingExpressionBuilder expressionBuilder;
+    private IDocument document;
+    private Expression generatedExpression;
+    private final GroovySourceViewerFactory sourceViewerFactory;
+    private RootContractInputGenerator rootContractInputGenerator;
+    private Text scriptNameText;
+
+    /**
+     * @param pageName
+     */
+    public GeneratedScriptPreviewPage(final WritableValue rootNameObservable, final WritableList fieldToContactInputMappingsObservable,
+            final WritableValue selectedDataObservable, final RepositoryAccessor repositoryAccessor,
+            final FieldToContractInputMappingOperationBuilder operationBuilder, final FieldToContractInputMappingExpressionBuilder expressionBuilder,
+            final GroovySourceViewerFactory sourceViewerFactory) {
+        super(GeneratedScriptPreviewPage.class.getName());
+        setTitle(Messages.generatedScriptPreviewTitle);
+        this.rootNameObservable = rootNameObservable;
+        fieldToContractInputMappingsObservable = fieldToContactInputMappingsObservable;
+        this.selectedDataObservable = selectedDataObservable;
+        this.operationBuilder = operationBuilder;
+        this.expressionBuilder = expressionBuilder;
+        this.repositoryAccessor = repositoryAccessor;
+        this.sourceViewerFactory = sourceViewerFactory;
+        generatedExpression = ExpressionFactory.eINSTANCE.createExpression();
+    }
+
+    public void setDescription() {
+        if (selectedDataObservable.getValue() != null) {
+            setDescription(Messages.bind(Messages.setGeneratedScriptPreviewPageDescription, ((Element) selectedDataObservable.getValue()).getName()));
+            EMFObservables.observeDetailValue(Realm.getDefault(), selectedDataObservable, ProcessPackage.Literals.ELEMENT__NAME).addValueChangeListener(
```

Comment:
It is same code as the method setTitle of CreateContractInputFromBusinessObjectWizardPage